contracts/DEXes/UniV3StablesJoint.sol

```diff
@@ -0,0 +1,639 @@
+     */
+    function cloneUniV3StablesJoint(
+        address _providerA,
+        address _providerB,
+        address _referenceToken,
+        address _pool,
+        uint24 _ticksFromCurrent
+    ) external returns (address newJoint) {
+        require(isOriginal, "!original");
+        bytes20 addressBytes = bytes20(address(this));
+
+        assembly {
+            // EIP-1167 bytecode
+            let clone_code := mload(0x40)
+            mstore(
+                clone_code,
+                0x3d602d80600a3d3981f3363d3d373d3d3d363d73000000000000000000000000
+            )
+            mstore(add(clone_code, 0x14), addressBytes)
+            mstore(
+                add(clone_code, 0x28),
+                0x5af43d82803e903d91602b57fd5bf30000000000000000000000000000000000
+            )
+            newJoint := create(0, clone_code, 0x37)
+        }
+
+        UniV3StablesJoint(newJoint).initialize(
+            _providerA,
+            _providerB,
+            _referenceToken,
+            _pool,
+            _ticksFromCurrent
+        );
+
+        emit Cloned(newJoint);
+    }
+
+    /*
+     * @notice
+     *  Function returning the name of the joint in the format "NoHedgeUniV3StablesJoint(USDC-DAI)"
+     * @return name of the strategy
+     */
+    function name() external view override returns (string memory) {
+        string memory ab = string(
+            abi.encodePacked(
+                IERC20Extended(address(tokenA)).symbol(),
+                "-",
+                IERC20Extended(address(tokenB)).symbol()
+            )
+        );
+
+        return string(abi.encodePacked("NoHedgeUniV3StablesJoint(", ab, ")"));
+    }
+
+    /*
+     * @notice
+     *  Function returning the liquidity amount of the LP position
+     * @return liquidity from positionInfo
+     */
+    function balanceOfPool() public view override returns (uint256) {
+        IUniswapV3Pool.PositionInfo memory positionInfo = _positionInfo();
+        return positionInfo.liquidity;
+    }
+
+    /*
+     * @notice
+     *  Function available for vault managers to set the CRV pool to use for swaps
+     * @param newPool, new CRV pool address to use
+     */
+    function setCRVPool(address newPool) external onlyVaultManagers {
+        crvPool = newPool;
+    }
+
+    /*
+     * @notice
+     *  Function available for vault managers to set the boolean value deciding wether
+     * to use the uni v3 pool for swaps or a CRV pool
+     * @param newUseCRVPool, new boolean value to use
+     */
+    function setUseCRVPool(bool newUseCRVPool) external onlyVaultManagers {
+        useCRVPool = newUseCRVPool;
+    }
+
+    /*
+     * @notice
+     *  Function available for vault managers to set the number of ticks on each side of 
+     * current tick to provide liquidity to
+     * @param newTicksFromCurrent, new value to use
+     */
+    function setTicksFromCurrent(uint24 newTicksFromCurrent) external onlyVaultManagers {
```

Comment:
I would protect this function in case a position is open (as it could mess up with position retrieval).

it would mean to revert if a position is open AND a force argument is not passed.

Reply:
Ticks from current is only used to ope the position, I think it makes sense to leave it "unprotected" as we'd be needing to potentially change it before the current epoch ends to anticipate the # of ticks the next epoch provides liquidity to, it does not play a role in retrieving the current position as `minTick` and `maxTick`

Reply:
++ @16slim, you need to be able to set this before harvesting to close the previous epoch and open a new one.